src/Event/EventManager.php

```diff
@@ -86,7 +86,7 @@ public static function instance(?EventManager $manager = null): EventManager
         if ($manager instanceof EventManager) {
             static::$_generalManager = $manager;
         }
-        if (empty(static::$_generalManager)) {
+        if (!static::$_generalManager instanceof \Cake\Event\EventManager) {
```

Comment:
Could this be just
```suggestion
        if (static::$_generalManager === null) {
```
?